tests/15-rpl-classic/code-simple-tx-rx/simple-node.c

```diff
@@ -0,0 +1,112 @@
+#include "contiki.h"
+#include "contiki-lib.h"
+#include "contiki-net.h"
+#include "net/routing/routing.h"
+#include "node-id.h"
+#include <stdint.h>
+
+/*---------------------------------------------------------------------------*/
+#include "sys/log.h"
+#define LOG_MODULE "App"
+#define LOG_LEVEL LOG_LEVEL_INFO
+/*---------------------------------------------------------------------------*/
+
+#define UDP_PORT      61618
+#define PAYLOAD_SIZE  100
+#define TX_INTERVAL   (10 * CLOCK_SECOND)
+#define TX_WAIT       (60 * CLOCK_SECOND)
+
+#ifdef SIMPLE_CONF_TX_COUNT_MAX
+#define SIMPLE_TX_COUNT_MAX  SIMPLE_CONF_TX_COUNT_MAX
+#else
+#define SIMPLE_TX_COUNT_MAX  1000
+#endif
+
+#ifdef SIMPLE_CONF_NODE_TX_ID
+#define SIMPLE_NODE_TX_ID    SIMPLE_CONF_NODE_TX_ID
+#else
+#define SIMPLE_NODE_TX_ID    2
+#endif
+
+#ifdef SIMPLE_CONF_NODE_RX_ID
+#define SIMPLE_NODE_RX_ID    SIMPLE_CONF_NODE_RX_ID
+#else
+#define SIMPLE_NODE_RX_ID    1
+#endif
+
+static struct simple_udp_connection unicast_connection;
+
+/*---------------------------------------------------------------------------*/
+PROCESS(udp_process, "UDP unicast process");
+AUTOSTART_PROCESSES(&udp_process);
+/*---------------------------------------------------------------------------*/
+static void
+receiver(struct simple_udp_connection *c,
+         const uip_ipaddr_t *sender_addr,
+         uint16_t sender_port,
+         const uip_ipaddr_t *receiver_addr,
+         uint16_t receiver_port,
+         const uint8_t *data,
+         uint16_t datalen)
+{
+  static uint32_t rx_count = 0;
+  rx_count++;
+  LOG_INFO("Data %u received on port %d from port %d with length %d\n",
```

Comment:
Use the `PRI` family of defines from `inttypes.h` to print fixed size types. (`PRIu32`/`PRIu16` for these).

Reply:
Fixed in #2548